### DISCUSSION TITLE
how to customize preview UX similar to picker UX

### DISCUSSION THREAD

**Discussion by rbarzel** [fluttercandies/flutter_wechat_assets_picker#507] [Q&A]
This is a very impressive package!!!

I'm able to successfully customize the picker UX by passing my custom `AssetPickerBuilderDelegate` into `AssetPicker.pickAssetsWithDelegate()`. 

But I can't figure out how to customize the preview UX. (See an example of the screen I want to customize below.)

I think I need to create a custom `AssetPickerViewerBuilderDelegate`, but I don't' understand how to tell AssetPicker to use my custom `AssetPickerViewerBuilderDelegate`.

I'd appreciate any guidance. Thanks!

![Screenshot_20231113-131258](https://github.com/fluttercandies/flutter_wechat_assets_picker/assets/1355649/7c15a7a3-d47d-40d2-a9e0-fd7eee0c4b24)

**Comment by AlexV525** [comment]
- https://pub.dev/documentation/wechat_assets_picker/latest/wechat_assets_picker/AssetPickerBuilderDelegate/viewAsset.html
- https://github.com/fluttercandies/flutter_wechat_assets_picker/blob/0015480e385a59a9188c6cba7f329a96a9245ff1/lib/src/delegates/asset_picker_builder_delegate.dart#L871

**Reply by rbarzel** [reply]
Thanks for the quick reply @AlexV525.  It seems I can customize the preview of an individual asset by overriding the `AssetPickerBuilderDelegate.viewAsset`. But `viewAsset` is not called when I click **Preview** to view all selected assets, and that's the screen I'm trying to customize. 

I'm guessing that I need to override `AssetPickerBuilderDelegate.previewButton`, and in my override code I should call  [AssetPickerViewer.pushToViewerWithDelegate](https://github.com/fluttercandies/flutter_wechat_assets_picker/blob/0015480e385a59a9188c6cba7f329a96a9245ff1/lib/src/widget/asset_picker_viewer.dart#L82C45-L82C45) and provide my own `AssetPickerViewerBuilderDelegate`. Is this the right approach?

**Reply by AlexV525** [reply]
I'll check with the preview button later.

**Comment by rbarzel** [comment]
I was able to customize the preview screen by following the suggestion of @AlexV525 to override `DefaultAssetPickerBuilderDelegate.viewAsset` and inside that method call `AssetPickerViewer.pushToViewerWithDelegate` using my own implementation of `AssetPickerViewerBuilderDelegate`. 

But this seems like a brittle solution because when the source repo implementation of `viewAsset` changes my (customized) version will be out of date. It would be great if the `DefaultAssetPickerBuilderDelegate` constructor had an optional argument like `viewerBuilder = DefaultAssetPickerViewerBuilderDelegate`. And `viewAsset` would always call `pushToViewerWithDelegate`.

Thanks for the help and for the great package!
